1 table.sql

```diff
@@ -0,0 +1,59 @@
+use FlyAirDB
+go
+drop table if exists Booking
+go 
+create table dbo.Booking(
+    BookingId int not null identity primary key,
+    FlightNum char(6) not null 
+        constraint ck_Booking_flight_number_must_start_with_Fly_then_fllowed_by_3_numbers check(FlightNum like 'Fly[0-9][0-9][0-9]'),
+    DepartureAirport char(3) not null
+        constraint ck_Booking_departure_airport_must_be_3_letters check(DepartureAirport like '[a-z][a-z][a-z]'),
+    DepartureCountry varchar(25) not null
+        constraint ck_Booking_departure_country_cannot_be_blank check(DepartureCountry <> ''),
+    DepartureTime datetime not null,
+    ArrivalAirport char(3) not null
+        constraint ck_Booking_arrival_airport_must_be_3_letters check(ArrivalAirport like '[a-z][a-z][a-z]'),
+    ArrivalCountry varchar(25) not null
+        constraint ck_Booking_arrival_country_cannot_be_blank check(ArrivalCountry <> ''),
+    ArrivalTime datetime not null,
+    PassengerName varchar(50) not null
+        constraint ck_Booking_passenger_name_cannot_be_blank check(PassengerName <> ''),
+    PassengerDOB date not null,
+    PassengerAddress varchar(100) not null
+        constraint ck_Booking_passenger_address_cannot_be_blank check(PassengerAddress <> ''),
+    BookedDate date null,
+    PassportNum varchar(9) null
+        constraint ck_Booking_passport_num_cannot_start_with_0 check(PassportNum not like '0%'),
+        constraint ck_Booking_passport_num_length_must_be_9 check(len(PassportNum) = 9),
+        constraint ck_Booking_passport_num_can_only_be_numbers check(PassportNum not like '%[^0-9]%'),
+    PassportIssueDate date null,
+    PassportExpiryDate as case 
```

Comment:
The case should be a constraint. It doesn't make sense to have expiry date as computed column.

Reply:
Because expiry date is data you add to the ticket. And it has to do with the passport not with the ticket. It's not necessarily based on the other dates.

On a side note. If you have a passport that's still valid, and you make a new one, you will have extra dates on your passport but if it's > 10 years old at time of travel, you won't be able to travel even if it's still valid.